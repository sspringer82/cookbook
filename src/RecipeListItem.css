.RecipeListItemRow {
  display: flex;
  line-height: 40px;
  margin: 0 20px;
  box-shadow: 2px 2px 5px 0 rgba(150, 150, 150, 0.75);
}

.RecipeListItemTitle {
  min-width: 200px;
}

.RecipeListItemPadding {
  padding-left: 20px;
}
