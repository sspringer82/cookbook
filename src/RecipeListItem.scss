.RecipeListItem {
  .Row {
    display: flex;
    line-height: 40px;
    margin: 0 20px;
    box-shadow: 2px 2px 5px 0 rgba(150, 150, 150, 0.75);
  }

  .Title {
    min-width: 200px;
  }

  .Padding {
    padding-left: 20px;
  }
}
