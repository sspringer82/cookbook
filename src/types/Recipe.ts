export type Recipe = {
  id: number;
  title: string;
  ingredients: Ingredient[];
  steps: string[];
  rating: number;
  image?: string;
};

export type Ingredient = {
  title: string;
  amount: number;
  unit: string;
};
