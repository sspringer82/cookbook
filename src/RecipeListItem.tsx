import React, { useContext } from 'react';
import { Recipe } from './types/Recipe';
import { Padding, Row, Title } from './RecipeListItem.styles';
import DeleteIcon from '@material-ui/icons/Delete';
import IconButton from '@material-ui/core/IconButton';
import { darkModeContext } from './darkModeContext';
import { Link } from 'react-router-dom';

type Props = {
  recipe: Recipe;
  onDelete: (id: number) => void;
};

function RecipeListItem({ recipe, onDelete }: Props): React.ReactElement {
  const [darkMode] = useContext(darkModeContext);

  // useEffect(() => {
  //   return () => {
  //     console.log('Component mit der ID ', recipe.id, ' wurde entfernt');
  //   };
  // }, [recipe.id]);

  return (
    <Row>
      <Title darkMode={darkMode} data-testid="title">
        <Link to={`/detail/${recipe.id}`}>{recipe.title}</Link>
      </Title>
      <Padding darkMode={darkMode}>
        <IconButton
          aria-label="delete"
          onClick={() => {
            onDelete(recipe.id);
          }}
          data-testid="deleteButton"
        >
          <DeleteIcon />
        </IconButton>
      </Padding>
    </Row>
  );
}

export default RecipeListItem;
