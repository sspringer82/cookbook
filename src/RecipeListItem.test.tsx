import { fireEvent, render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import RecipeListItem from './RecipeListItem';
import { Recipe } from './types/Recipe';

const recipe: Recipe = {
  id: 1,
  title: 'Nudeln',
  ingredients: [{ title: 'Mehl', amount: 1, unit: 'kg' }],
  steps: ['machen'],
};

describe('RecipeListItem', () => {
  it('should render', () => {
    render(
      <MemoryRouter>
        <RecipeListItem recipe={recipe} onDelete={jest.fn()} />
      </MemoryRouter>,
    );

    expect(screen.getByText('Nudeln')).toBeInTheDocument();
    expect(screen.getByTestId('title')).toHaveTextContent('Nudeln');
  });

  it('should delete the Recipe', () => {
    const deleteSpy = jest.fn();
    render(
      <MemoryRouter>
        <RecipeListItem recipe={recipe} onDelete={deleteSpy} />
      </MemoryRouter>,
    );

    fireEvent.click(screen.getByTestId('deleteButton'));

    expect(deleteSpy).toHaveBeenCalledWith(1);
  });

  it('should match a snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <RecipeListItem recipe={recipe} onDelete={jest.fn()} />
      </MemoryRouter>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
