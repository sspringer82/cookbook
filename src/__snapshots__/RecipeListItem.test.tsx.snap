// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeListItem should match a snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiCard-root sc-bdnxRM exgsRF MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="sc-gtsrHT sc-dlnjwi brqfbx hiIKPJ"
      data-testid="title"
    >
      <a
        href="/detail/1"
      >
        Nudeln
      </a>
    </div>
    <div
      class="sc-gtsrHT brqfbx"
    >
      <button
        aria-label="delete"
        class="MuiButtonBase-root MuiIconButton-root"
        data-testid="deleteButton"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
